File: Python version/platforms/desktop/desktop_handler.py
from base.handler import Handler


class DesktopHandler(Handler):
    """Handler specialization for desktop platforms."""

    def __init__(self, service):
        super(DesktopHandler, self).__init__()
        self.__service = service

    def start_async(self, c, program_index=None, option_index=None):
        """Specialization of superclass start_async method.

        Starts ASP solving asynchronously a subset of data and
        options for a desktop platform.
        """
        input_programs = self._collect_programs(program_index)
        input_options = self._collect_options(option_index)
        self.__service.start_async(c, input_programs, input_options)

    def start_sync(self, program_index=None, option_index=None):
        """Specialization of superclass start_sync method.

        Starts ASP solving synchronously a subset of data and options
        for a desktop platform.
        """
        input_programs = self._collect_programs(program_index)
        input_options = self._collect_options(option_index)
        return self.__service.start_sync(input_programs, input_options)

File: CSharp version/it/unical/mat/embasp/platforms/desktop/DesktopHandler.cs
using System.Collections.Generic;

namespace it.unical.mat.embasp.platforms.desktop
{
    using ICallback = it.unical.mat.embasp.@base.ICallback;
    using Handler = it.unical.mat.embasp.@base.Handler;
    using InputProgram = it.unical.mat.embasp.@base.InputProgram;
    using OptionDescriptor = it.unical.mat.embasp.@base.OptionDescriptor;
    using Output = it.unical.mat.embasp.@base.Output;

    public class DesktopHandler : Handler
    {
        private readonly DesktopService service;

        public DesktopHandler(DesktopService service) => this.service = service;

        public override void StartAsync(ICallback c, IList<int> program_index, IList<int> option_index)
        {
            IList<InputProgram> input_programs = CollectPrograms(program_index);
            IList<OptionDescriptor> input_options = CollectOptions(option_index);
            service.StartAsync(c, input_programs, input_options);
        }

        public override Output StartSync(IList<int> program_index, IList<int> option_index)
        {
            IList<InputProgram> input_programs = CollectPrograms(program_index);
            IList<OptionDescriptor> input_options = CollectOptions(option_index);
            return service.StartSync(input_programs, input_options);
        }
    }
}